### PR TITLE
Add beatmap database

### PR DESCRIPTION
**Pull request by ddevault**
This just does the basics, it lays out the schema and initializes the db. More to follow in later PRs. This also contains the changes necessary following ppy/osu-framework#108.

### CODE REVIEW

File: osu.Game/Beatmaps/GameMode.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+namespace osu.Game.Beatmaps
+{
+    public enum GameMode
```
Comment (peppy) [inline osu.Game/Beatmaps/GameMode.cs]: PlayMode exists in my PR #43

Comment (SirCmpwn) [inline osu.Game/Beatmaps/GameMode.cs]: Noted. Let's get #43 merged first, then.

File: osu.Game/Database/BeatmapDatabase.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using osu.Framework.OS;
+using SQLite;
+
+namespace osu.Game.Database
+{
+    public class BeatmapDatabase
+    {
+        private static SQLiteConnection Connection { get; set; }
```
Comment (peppy) [inline osu.Game/Database/BeatmapDatabase.cs]: feels like this almost doesn't need to be static?

File: osu.Game/Database/BeatmapSet.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using SQLite;
+
+namespace osu.Game.Database
+{
+    public class BeatmapSet
+    {
+        [PrimaryKey]
+        public int BeatmapSetID { get; set; }
+        [NotNull, Indexed]
+        public int MetadataID { get; set; }
```
Comment (peppy) [inline osu.Game/Database/BeatmapSet.cs]: BeatmapMetadataID to be in-line with other column name

File: osu.Game/Database/BeatmapDatabase.cs (hunk 2 above)
Comment (SirCmpwn) [inline osu.Game/Database/BeatmapDatabase.cs]: No, you can't have the same beatmap file open multiple times. It's also not thread safe.

File: osu.Desktop.VisualTests/Program.cs [hunk 4]
```diff
@@ -12,7 +12,7 @@ public static class Program
         [STAThread]
         public static void Main(string[] args)
         {
-            BasicGameHost host = Host.GetSuitableHost();
+            BasicGameHost host = Host.GetSuitableHost("osu-tests");
```
Comment (SirCmpwn) [inline osu.Desktop.VisualTests/Program.cs]: @"osu-tests"

File: osu.Desktop/Program.cs [hunk 5]
```diff
@@ -11,9 +11,9 @@ namespace osu.Desktop
     public static class Program
     {
         [STAThread]
-        public static void Main()
+        public static void Main(string[] args)
         {
-            BasicGameHost host = Host.GetSuitableHost();
+            BasicGameHost host = Host.GetSuitableHost("osu");
```
Comment (SirCmpwn) [inline osu.Desktop/Program.cs]: @"osu"

**Comment by SirCmpwn** [comment]
Ditching this PR in favor of #45

**Pull request closed without merging**